woocommerce/woocommerce
# [e2e tests] Add support for external sites in CI

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/48920
Part of pdV5qK-wB-p2

Add support for running e2e tests against any site in CI.

- added scripts that encrypt and decrypt .env files.
- added a new env entry - `default-pressable` to validate the setup.
- configured the `Core e2e tests - default Pressable site` job to run on `daily-checks`.

### How to test the changes in this Pull Request:

Locally:

Run the tests with `pnpm test:e2e:with-env default-pressable`. You'll need to find and set the `E2E_ENV_KEY` for this.
Expected:
- The `tests/e2e-pw/envs/default-pressable/.env.enc` is decrypted into `tests/e2e-pw/.env`
- The `tests/e2e-pw/envs/default-pressable/env-setup.sh` script is executed. It doesn't do anything (yet)
- The tests configured in `tests/e2e-pw/envs/default-pressable/playwright.config.js` should run

In CI:
Check the `Core e2e tests - default Pressable site` job: https://github.com/woocommerce/woocommerce/actions/runs/9782549328/job/27009409289?pr=49017. It was temporarily configured to run for PRs to validate the CI setup.
